In [1]:
import pandas as pd
dfmortality = pd.read_csv("D:/Assignments/SEM 2/Healthcare/Project/Mortality Data.csv")
filter = dfmortality[dfmortality.ICD10.str.contains('J')]
filter.sort_values("ICD10",inplace = True,ascending = True)

D:\Softwares\Anaconda\lib\site-packages\ipykernel_launcher.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  after removing the cwd from sys.path.


In [2]:
dfmortality.columns

Index(['CountryName', 'MortalityYear', 'Gender', 'AgeGroupCode', 'ICD10',
       'Deaths'],
      dtype='object')

Q1 Mortality rate per country over the years due to respiratory diseases
Analysis to be derived : Which country has the highest average Mortality rate over the years and Why?
(Population Dataset of every country needs to be imported to find this out) --> Needs to be discussed

In [3]:
groupedfilter = filter.groupby(['CountryName','MortalityYear']).size().reset_index()
print(groupedfilter)

                          CountryName  MortalityYear     0
0                            Anguilla           2014     2
1                            Anguilla           2016     2
2                            Anguilla           2017     4
3                            Anguilla           2018     1
4                 Antigua and Barbuda           2013    13
5                 Antigua and Barbuda           2014    29
6                 Antigua and Barbuda           2015    24
7                 Antigua and Barbuda           2016    15
8                 Antigua and Barbuda           2017    29
9                           Argentina           2011    55
10                          Argentina           2013   468
11                          Argentina           2014  1326
12                          Argentina           2015   590
13                          Argentina           2016   462
14                          Argentina           2017  1453
15                              Aruba           2013    

Q2 No of deaths per disease to identify the most and least dangerous disease
(Line graph to be created to map no of deaths vs disease)

In [4]:
groupedfilter= filter.groupby(['ICD10']).size().reset_index()
groupedfilter.columns = ['ICD10','Deaths']
groupedfilter.sort_values("Deaths",inplace = True,ascending = False)
print(groupedfilter)

    ICD10  Deaths
56   J189    2951
52   J180    2297
164  J690    2101
126  J449    1891
130  J459    1885
175  J81     1824
178  J841    1658
209  J969    1602
214  J984    1551
46   J159    1518
207  J960    1396
123  J440    1368
53   J181    1319
217  J988    1250
187  J90     1197
122  J439    1175
69   J22     1175
174  J80     1145
180  J849    1054
124  J441    1028
131  J46     1027
125  J448     940
117  J42      919
113  J40      903
132  J47      868
54   J182     836
64   J209     806
186  J869     794
208  J961     704
210  J980     703
..    ...     ...
203  J954       6
199  J950       5
62   J206       5
66   J211       5
101  J382       5
59   J202       5
137  J630       5
151  J671       5
93   J353       5
201  J952       5
206  J959       4
4    J013       4
219  J998       4
60   J204       3
3    J012       3
87   J342       3
51   J178       3
153  J674       3
49   J170       3
70   J303       2
86   J341       2
148  J662       2
88   J343       2
50   J172 

From the above analysis we identify :
1. Most dangerous disease : J189 with 2951 deaths
Q.3 Countries affected by J189(Pneumonia)
Analysis to be derived : Why Argentina was most affected by J189?(Predictive Modeling could be used to identify external factors like GDP,Medical conditions, climate impacting the high mortality rate here)
(Bar Graph needs to be created to depict the frequency of breakout per country)

In [5]:
maxBreakout= filter[filter["ICD10"] == "J189"].groupby('CountryName').size().reset_index()
maxBreakout.columns = ['CountryName','Deaths']
maxBreakout.sort_values("Deaths",inplace = True,ascending = False)
print(maxBreakout)

                         CountryName  Deaths
2                          Argentina     187
12                          Colombia     186
32                              Peru     185
29                         Nicaragua     184
8                            Brazil      173
27                            Mexico     151
22                         Guatemala     146
40          United States of America     142
31                          Paraguay     140
30                            Panama     131
9                             Canada     125
18                       El Salvador     123
11                             Chile     115
41                           Uruguay     113
14                              Cuba     108
33                       Puerto Rico      89
42                         Venezuela      68
17                           Ecuador      68
6                             Belize      45
25                           Jamaica      44
23                            Guyana      43
35  Saint 

Q4 Identified the age most affected by J189(Pneumonia) - Age 27
Identified the age least affected by J189(Pneumonia) - Age 31
(Draw a pie chart with age groups distribution)
Analysis to be derived : Why J189 was most prevalent in this age group? (Predictive modeling could be used with input of external factors like smoking habits,work conditions to predict the age distribution)

In [6]:
ageDist= filter[filter["ICD10"] == "J189"].groupby(['AgeGroupCode']).size().reset_index()
ageDist.columns = ['Age','Deaths']
ageDist.sort_values("Deaths",inplace = True,ascending = False)
print(ageDist)

    Age  Deaths
25   27     181
24   26     168
26   28     164
23   25     154
17   19     153
19   21     145
22   24     144
18   20     141
21   23     138
16   18     130
20   22     128
15   17     126
3     5     121
14   16     114
4     6     100
5     7      88
8    10      86
13   15      86
11   13      83
10   12      81
12   14      80
6     8      78
9    11      74
7     9      72
27   29      57
2     4      32
30   32       7
1     3       7
0     2       6
28   30       4
29   31       3


Q5 Identified the gender most affected by J189(Pneumonia)
Result : Females are most impacted by J189
Analysis to be derived : Causes for the gender being most affected by J189

In [7]:
genderDist= filter[filter["ICD10"] == "J189"].groupby(['Gender']).size().reset_index()
genderDist.columns = ['Gender','Deaths']
genderDist.sort_values("Deaths",inplace = True,ascending = False)
print(ageDist)

    Age  Deaths
25   27     181
24   26     168
26   28     164
23   25     154
17   19     153
19   21     145
22   24     144
18   20     141
21   23     138
16   18     130
20   22     128
15   17     126
3     5     121
14   16     114
4     6     100
5     7      88
8    10      86
13   15      86
11   13      83
10   12      81
12   14      80
6     8      78
9    11      74
7     9      72
27   29      57
2     4      32
30   32       7
1     3       7
0     2       6
28   30       4
29   31       3


Q6 Time trend of Argentina ie. outbreak of disease over the period
(Histogram needs to be created to perform mapping of time interval and Year)

In [9]:
timeTrend = filter[(filter["ICD10"] == "J189") & (filter["CountryName"] == "Argentina")].groupby(['MortalityYear']).size().reset_index()
print(timeTrend)

   MortalityYear   0
0           2011   6
1           2013  13
2           2014  59
3           2015  24
4           2016  21
5           2017  64
